quadruped_pympc/controllers/Controllers.py

```diff
@@ -0,0 +1,543 @@
+                 swing_generator=None,
+                 mpc_frequency= None,
+                 mpc_dt=None,
+                 controller_type=None,
+                 horizon=None,
+                 optimize_step_freq=None,
+                 step_freq_available=None,
+                 sampling_method=None,
+                 control_parametrization=None,
+                 sigma_cem_mppi=None,
+                 num_parallel_computations=None,
+                 num_sampling_iterations=None,
+                 use_nonuniform_discretization=False,
+                 shift_solution=None,
+                 dt_fine_grained=None,
+                 horizon_fine_grained=None,
+                 use_inertia_recomputation=None,
+                 inertia=None):
+        #
+        self.env=env
+        self.gait_name=gait_name
+        self.simulation_dt = sim_dt
+        self.hip_height = hip_height
+        self.legs_order = legs_order
+        self.ref_z=hip_height
+        self.inertia=inertia
+        self.use_inertia_recomputation=use_inertia_recomputation
+        self.sigma_cem_mppi=sigma_cem_mppi
+        self.use_nonuniform_discretization=use_nonuniform_discretization
+
+        ### All this parameter are for the WBC 
+        self.step_height = step_height
+        self.position_gain_fb = swing_position_gain_fb
+        self.velocity_gain_fb = swing_velocity_gain_fb
+        self.swing_generator = swing_generator
+
+        # MPC parameters
+        self.mpc_frequency = mpc_frequency
+        self.mpc_dt = mpc_dt
+        self.horizon = horizon
+        self.controller_type= controller_type
+        self.optimize_step_frequency= optimize_step_freq
+        self.step_freq_available = step_freq_available
+        if self.controller_type == 'sampling':
+            sampling_method = sampling_method
+            control_parametrization = control_parametrization
+            num_parallel_computations = num_parallel_computations
+            self.iteration = num_sampling_iterations
+            shift_solution = shift_solution
+        
+        if use_nonuniform_discretization==True :
+            dt_fine_grained = dt_fine_grained
+            horizon_fine_grained = horizon_fine_grained
+
+
+        self.stance_time=0
+        self.swing_period = 0
+
+        self.nmpc_grf_old=None
+
+
+
+        # Initialization of variables used in the main control loop
+        # ____________________________________________________________
+        # Set the reference for the state
+        self.ref_pose = np.array([0, 0, hip_height])
+        self.ref_base_lin_vel, self.ref_base_ang_vel = env.target_base_vel()
+        self.ref_orientation = np.array([0.0, 0.0, 0.0])
+        # # SET REFERENCE AS DICTIONARY
+        # TODO: I would suggest to create a DataClass for "BaseConfig" used in the PotatoModel controllers.
+        self.ref_state = {}
+        # Starting contact sequence
+        self.previous_contact = np.array([1, 1, 1, 1])
```

Comment:
@giulioturrisi 

why these two instances?

Reply:
which one? previous_contact_sequence? to keep track of the switch between stance and swing.

BTW, can we use all lower caps for the name of the file? the class can be capital letters